### PR TITLE
Fixes #1388: implement TLS support in the tcp_lite adaptor

### LINKED ISSUE #1388: Implement TLS endpoint termination for TCP lite adaptor

### PR DESCRIPTION
**Pull request by kgiusti**
Closes #1388

### CODE REVIEW

File: src/adaptors/adaptor_listener.c [hunk 1]
```diff
@@ -152,12 +150,9 @@ static void _listener_event_handler(pn_event_t *e, qd_server_t *qd_server, void
...
             else {
-                // the adaptor_listener is closing but a connection attempt arrived before the
-                // cleanup is complete.  Need to accept it then close it when the connection
-                // completes
-                pn_raw_connection_t *close_me = pn_raw_connection();
-                pn_raw_connection_set_context(close_me, &_conn_event_context);
-                pn_listener_raw_accept(pn_event_listener(e), close_me);
+                // The adaptor_listener is closing but a connection attempt arrived before the
+                // cleanup is complete. Deny the connection attempt.
+                qd_adaptor_listener_deny_conn(li, pn_event_listener(e));
```
Comment (ganeshmurthy) [inline src/adaptors/adaptor_listener.c:155]: Above there is a debug log message that says new incoming client connetion. When we explicitly add another log message when we deny the connection in the _deny_conn_handler ?

File: src/adaptors/adaptor_listener.c [hunk 2]
```diff
@@ -152,12 +150,9 @@ static void _listener_event_handler(pn_event_t *e, qd_server_t *qd_server, void
             if (li->on_accept)
                 li->on_accept(li, pn_event_listener(e), li->user_context);
             else {
-                // the adaptor_listener is closing but a connection attempt arrived before the
-                // cleanup is complete.  Need to accept it then close it when the connection
-                // completes
-                pn_raw_connection_t *close_me = pn_raw_connection();
-                pn_raw_connection_set_context(close_me, &_conn_event_context);
-                pn_listener_raw_accept(pn_event_listener(e), close_me);
+                // The adaptor_listener is closing but a connection attempt arrived before the
```
Comment (ganeshmurthy) [inline src/adaptors/adaptor_listener.c:153]: We don't deny connection in the handler of the PN_LISTENER_ACCEPT event in server.c
Should we ?

File: src/adaptors/adaptor_listener.c (hunk 1 above)
Comment (kgiusti) [reply src/adaptors/adaptor_listener.c:155]: Sure... coming up.

File: src/adaptors/adaptor_listener.c (hunk 2 above)
Comment (kgiusti) [reply src/adaptors/adaptor_listener.c:153]: Hrmmm... that server listener has a different lifecycle than the adaptor listener: it stays active until the listener config is explicitly deleted.  That's unlike an adaptor listner which needs to disable/enable the socket frequently depending on the availability of a consumer...

So my feeling is no - this won't happen with server listeners.

File: src/adaptors/adaptor_listener.c (hunk 1 above)
Comment (kgiusti) [reply src/adaptors/adaptor_listener.c]: Added with latest commit.

**Comment by kgiusti** [comment]
The last outstanding test failure - in the Http/1-over-tcp test - appears to be unrelated to the TLS implementation. I've opened a separate issue for tracking that:

https://github.com/skupperproject/skupper-router/issues/1432

I think the reason we're seeing that failure is because this patch enables the Http/1-over-tcp test which triggered the issue.

I'm going to merge this patch and turn my attention to #1432 next.

**Pull request merged**